src/js/main.js

```diff
@@ -28,20 +32,40 @@
     .when('/trip/:id/city', {
       templateUrl: 'timeline.html',
       controller: function($http, $scope, $location, $routeParams){
+
       $http.get( BASE_URL + '/api/trip/' + $routeParams.id + '/city/')
         .then(function (response){
-          console.log(arguments);
           // $scope.main = response.data;
           $scope.cities = response.data;
+          var waypoints = response.data;
+          console.log($scope.cities);
+          for (var i in waypoints) {
```

Comment:
The `for (key in object)` construct you're using here is really reserved for enumerating keys of a dictionary, and it's quite buggy. Better to use `forEach` instead, although if you're just picking keys out of each dictionary in a list, see `lodash.pick`.